shotgun_api3/lib/mockgun.py

```diff
@@ -690,36 +675,91 @@ def _get_field_type(self, entity_type, field):
 
-    def _row_matches_filter(self, entity_type, row, filter):
-        
-        
+    def _row_matches_filter(self, entity_type, row, filter, retired_only):
+
         try:
             field, operator, rval = filter
         except ValueError:
             raise ShotgunError("Filters must be in the form [lval, operator, rval]")
-        lval = self._get_field_from_row(entity_type, row, field)
-        
-        field_type = self._get_field_type(entity_type, field)
-        
-        # if we're operating on an entity, we'll need to grab the name from the lval's row
-        if field_type == "entity":
-            lval_row = self._db[lval["type"]][lval["id"]]
-            if "name" in lval_row:
-                lval["name"] = lval_row["name"]
-            elif "code" in lval_row:
-                lval["name"] = lval_row["code"]
-        return self._compare(field_type, lval, operator, rval)
+
+        # Special case, field is None when we have a filter operator.
+        if field is None:
+            if operator in ["any", "all"]:
+                return self._row_matches_filters(entity_type, row, rval, operator, retired_only)
+            else:
+                raise ShotgunError("Unknown filter_operator type: %s" % operator)
+        else:
+            lval = self._get_field_from_row(entity_type, row, field)
+
+            field_type = self._get_field_type(entity_type, field)
+
+            # if we're operating on an entity, we'll need to grab the name from the lval's row
+            if field_type == "entity":
+                lval_row = self._db[lval["type"]][lval["id"]]
+                if "name" in lval_row:
+                    lval["name"] = lval_row["name"]
+                elif "code" in lval_row:
+                    lval["name"] = lval_row["code"]
+            return self._compare(field_type, lval, operator, rval)
+
+    def _rearrange_filters(self, filters):
+        """
+        Modifies the filter syntax to turn it into a list of three items regardless
+        of the actual filter. Most of the filters are list of three elements, so this doesn't change much.
+
+        The filter_operator syntax uses a dictionary with two keys, "filters" and
+        "filter_operator". Filters using this syntax will be turned into
+        [None, filter["filter_operator"], filter["filters"]]
+
+        Filters of the form [field, operator, values....] will be turned into
+        [field, operator, [values...]].
+
+        :param list filters: List of filters to rearrange.
+
+        :returns: A list of three items.
+        """
+        rearranged_filters = []
+
+        # now translate ["field", "in", 2,3,4] --> ["field", "in", [2, 3, 4]]
+        for f in filters:
+            if isinstance(f, list):
+                if len(f) > 3:
+                    # ["field", "in", 2,3,4] --> ["field", "in", [2, 3, 4]]
+                    new_filter = [f[0], f[1], f[2:]]
+
+                elif f[1] == "in" and not isinstance(f[2], list):
```

Comment:
This is code that I have moved around, I've actually only added the `elif isinstance(f, dict)` branch. Honestly, the fact that I've just written the first unit tests for mockgun makes me a bit hesitant to update too much.